src/imaCore.mjs

```diff
@@ -1484,52 +1192,40 @@ export async function doAllS2S( // s-chain --> s-chain
     transactionCustomizerDst
 ) {
     let cntOK = 0, cntFail = 0, nIndexS2S = 0;
+    const sc = imaState.chainProperties.sc;
     const strDirection = "S2S";
     const arrSChainsCached = skaleObserver.getLastCachedSChains();
     const cntSChains = arrSChainsCached.length;
-    if( log.verboseGet() >= log.verboseReversed().information ) {
-        log.write( cc.debug( "Have " ) + cc.info( cntSChains ) +
-            cc.debug( " S-Chain(s) connected to this S-Chain for performing S2S transfers in " ) +
-            threadInfo.threadDescription() + cc.debug( "." ) +
-            "\n" );
-    }
+    log.information( "Have {} S-Chain(s) connected to this S-Chain for performing " +
+        "S2S transfers in {}.", cntSChains, threadInfo.threadDescription() );
     for( let idxSChain = 0; idxSChain < cntSChains; ++ idxSChain ) {
         const joSChain = arrSChainsCached[idxSChain];
-        const urlSrc = skaleObserver.pickRandomSChainUrl( joSChain );
+        const joPickResult = skaleObserver.pickRandomSChainIndexAndNodeAndUrl( joSChain );
+        const urlSrc = joPickResult.strURL;
```

Comment:
pickRandomSChainUrl should be used instead

Reply:
Done